### PR TITLE
[fix]: [CI-3203]: s3 upload step make region mandatory

### PR DESCRIPTION
**Pull request by shubham149**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31604)

### CODE REVIEW

File: 330-ci-beans/src/main/java/io/harness/beans/steps/stepinfo/UploadToS3StepInfo.java [hunk 1]
```diff
@@ -68,7 +68,7 @@
 
   // plugin settings
   @ApiModelProperty(dataType = STRING_CLASSPATH) private ParameterField<String> endpoint;
-  @ApiModelProperty(dataType = STRING_CLASSPATH) private ParameterField<String> region;
+  @NotNull @ApiModelProperty(dataType = STRING_CLASSPATH) private ParameterField<String> region;
```
Comment (harshjain12) [inline 330-ci-beans/src/main/java/io/harness/beans/steps/stepinfo/UploadToS3StepInfo.java:71]: Also check resolution
We should pass mandatory as true during resolution

Comment (shubham149) [reply 330-ci-beans/src/main/java/io/harness/beans/steps/stepinfo/UploadToS3StepInfo.java:71]: It is already true.
`String region = resolveStringParameter("region", "S3Upload", identifier, stepInfo.getRegion(), true);`

**Comment by shubham149** [comment]
trigger messagecheck

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31604#-:-MygPL0R25zN8KkqaXmH:bnfp4nl)* status: 1. unresolved discussions.

**Review by harshjain12 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31604)* status: :shipit: complete! ✓

**Pull request merged**